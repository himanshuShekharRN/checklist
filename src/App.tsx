import React from 'react';
import 'react-native-gesture-handler';

import {Navigation} from './navigation';
import {Provider} from 'react-redux';
import {persistor, store} from './store';
import {PersistGate} from 'redux-persist/integration/react';

export const App: React.FC = () => {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <Navigation />
      </PersistGate>
    </Provider>
  );
};
