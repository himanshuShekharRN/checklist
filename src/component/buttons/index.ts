import {FloatingAddButton} from './floatingAddButton';
import {RoundedButton} from './roundedButton';

export {FloatingAddButton, RoundedButton};
