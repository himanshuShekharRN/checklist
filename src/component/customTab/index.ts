import {CustomTab} from './CustomTab';
import {FirstRoute} from './firstRoute';
import {SecondRoute} from './secondRoute';
import {ThirdRoute} from './thirdRoute';
import {FourthRoute} from './fourthRoute';
import {FifthRoute} from './fifthRoute';

export {
  CustomTab,
  FirstRoute,
  SecondRoute,
  ThirdRoute,
  FourthRoute,
  FifthRoute,
};
