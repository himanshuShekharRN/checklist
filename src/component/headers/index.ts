import {HeadersWithButton} from './headersWithButton/HeadersWithButton';

export {HeadersWithButton};
