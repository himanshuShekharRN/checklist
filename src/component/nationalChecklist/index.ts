import {NationalChecklist} from './NationalChecklist';

export {NationalChecklist};
