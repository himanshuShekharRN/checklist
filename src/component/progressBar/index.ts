import {LinearProgressBar} from './linearProgressBar';

export {LinearProgressBar};
