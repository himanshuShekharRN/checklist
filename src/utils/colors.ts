export const COLOR_BLACK = '#000000';
export const COLOR_WHITE = '#FFFFFF';
export const COLOR_NAVY_300 = '#ECEEF0';
