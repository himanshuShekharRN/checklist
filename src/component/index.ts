import {SwipableCards} from './swipableCards/SwipableCards';
import {SwipableActionButton} from './swipableActionButton/SwipableActionButton';
import {Card} from './cards';
import {MyCheckList} from './myChecklist';
import {Space} from './space';
import {PreDepartureList} from './preDepartureList';
import {Title} from './title';
import {HeadersWithButton} from './headers';
import {CustomSVG} from './customSVG';
import {CustomIcon} from './customIcon';
import {FloatingAddButton, RoundedButton} from './buttons';
import {ListCard} from './listCard';
import {LinearProgressBar} from './progressBar';
import {CustomTab} from './customTab';
import {DepartureDocCard} from './departureDocCard';
import {NationalChecklist} from './nationalChecklist';

export {
  FloatingAddButton,
  Card,
  MyCheckList,
  Space,
  PreDepartureList,
  Title,
  HeadersWithButton,
  CustomSVG,
  CustomIcon,
  RoundedButton,
  ListCard,
  LinearProgressBar,
  CustomTab,
  DepartureDocCard,
  NationalChecklist,
  SwipableActionButton,
  SwipableCards,
};
