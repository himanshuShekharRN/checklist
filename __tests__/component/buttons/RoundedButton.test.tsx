import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';

import {RoundedButton} from '../../../src/component';
import {COLOR_BLACK} from '../../../src/utils/colors';

describe('Rounded Button', () => {
  const onPressMock = jest.fn();
  const customBtnStyle = {backgroundColor: 'transparent', borderRadius: 20};
  const customBtnTitleStyle = {color: COLOR_BLACK};

  const roundedButtonProps = {
    title: 'Press Me',
    onPress: onPressMock,
    buttonStyle: customBtnStyle,
    buttonTitleStyle: customBtnTitleStyle,
  };

  const tree: any = render(<RoundedButton {...roundedButtonProps} />);

  test('renders correctly', () => {
    expect(tree.toJSON()).toBeTruthy();
  });

  test('renders button with the correct title', () => {
    render(<RoundedButton {...roundedButtonProps} />);

    const button = screen.getByText('Press Me');
    expect(button).toBeTruthy();
  });

  test('calls onPress when the button is pressed', () => {
    render(<RoundedButton {...roundedButtonProps} />);

    const button = screen.getByText('Press Me');
    fireEvent.press(button);

    expect(onPressMock).toHaveBeenCalled();
  });

  test('applies custom buttonTitleStyle', () => {
    render(<RoundedButton {...roundedButtonProps} />);

    const buttonText = screen.getByText('Press Me');
    expect(buttonText.props.style).toContainEqual(customBtnTitleStyle);
  });

  test('snapshot created successfully', () => {
    render(<RoundedButton {...roundedButtonProps} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
