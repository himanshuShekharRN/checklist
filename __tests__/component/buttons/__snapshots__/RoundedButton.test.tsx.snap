// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rounded Button snapshot created successfully 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#0093BB",
        "borderRadius": 30,
        "height": 45,
        "justifyContent": "center",
        "width": "33%",
      },
      {
        "backgroundColor": "transparent",
        "borderRadius": 20,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "Roboto-Bold",
          "fontSize": 14,
          "textAlign": "center",
        },
        {
          "color": "#000000",
        },
      ]
    }
  >
    Press Me
  </Text>
</View>
`;
